pep/app/services/theme.ts

```diff
@@ -43,30 +43,59 @@ export default class ThemeService extends Service {
      */
     updateTheme(newThemeId: ThemeId) {
         this.currentUser.updatePrefs({ [PreferenceKey.THEME]: newThemeId });
-        this.setup();
+        if (window.document.querySelector('#theme')) {
+            window.document.querySelector('#theme')?.setAttribute('href', this.currentTheme.cssPath);
+        } else {
+            const linkElement = window.document.createElement('link');
+            linkElement.setAttribute('rel', 'stylesheet');
+            linkElement.setAttribute('type', 'text/css');
+            linkElement.setAttribute('id', 'theme');
+            linkElement.setAttribute('href', this.currentTheme.cssPath);
+            window.document.head.appendChild(linkElement);
+        }
+        this.injectLinkColors();
     }
 
-    @dontRunInFastboot
-    injectLinkColors() {
-        const colors = this.currentTheme.colors.links;
-        const styles = `
-            .glosstip {
+    getCssRules(theme: Theme) {
+        const colors = theme.colors.links;
+        return [
+            `.glosstip {
                 background-color: ${Color(colors.glossary).alpha(0.075)};
-            }
-            .glosstip:hover {
+            }`,
+            ` .glosstip:hover {
                 color: ${colors.glossary};
-            }
-            .bibtip {
+            }`,
+            `.bibtip {
                 color: ${colors.bibliography}
-            }
-            .pgx {
+            }`,
+            `.pgx {
                 color: ${colors.pageReference}
-            }
-            .figuretip {
+            }`,
+            ` .figuretip {
                 color: ${colors.figure}
-            }
-        `;
-        this.headData.set('linkColors', styles);
+            }`
+        ];
+    }
+
+    /**
+     * Inject link colors into the linkColors stylesheet. This happens to the administrators can control the colors of the links in the documents
+     *
+     * @memberof ThemeService
+     */
+    @dontRunInFastboot
+    injectLinkColors() {
```

Comment:
@adistasio did we get any feedback/clarification on theme management yet? I'd love if we could get rid of the extra overhead/complexity of this type of stuff, if it all could just live in the SASS.